lib/hiera/backend.rb

```diff
@@ -15,7 +15,7 @@ def initialize(wrapped)
 
       def lookup(key, scope, order_override, resolution_type, context)
         Hiera.debug("Using Hiera 1.x backend API to access instance of class #{@wrapped.class.name}. Lookup recursion will not be detected")
-        value = @wrapped.lookup(key, scope, order_override, resolution_type)
+        value = @wrapped.lookup(key, scope, order_override, resolution_type.is_a?(Hash) ? :hash : resolution_type)
```

Comment:
The _strategy_ of the Hash (or _behavior_ in this case, since I wanted to adhere to Hiera naming conventions), is not the same thing as the _resolution_type_. The _behavior_ is either {{:native}}, {{:deep}}, or {{:deeper}} and extends how the _resolution_type_ {{:hash}} works.